app/presenters/user_analytics_presenter.rb

```diff
@@ -160,37 +123,64 @@ def trophy_stats
     }
   end
 
-  #
-  # Groups by gender+date in the following format,
-  #
-  # { :grouped_by_gender_and_date =>
-  #    { "male" =>
-  #        { "Sun, 01 Mar 2020" => 21 },
-  #          "Tue, 01 Oct 2019" => 0 },
-  #      "female" =>
-  #        { "Sun, 01 Mar 2020" => 18,
-  #          "Tue, 01 Oct 2019" => 0 } } }
-  #
-  def group_by_gender_and_date(resource_data)
-    resource_data.inject(autovivified_hash) do |by_gender_and_date, (group, resource)|
-      gender, date = *group
-      by_gender_and_date[gender][date] = resource
-      by_gender_and_date
-    end
+  def daily_follow_ups
+    @current_facility
+      .patient_follow_ups(:day, last: DAYS_AGO)
+  end
+
+  def daily_registrations
+    @current_facility
+      .registered_patients
+      .group_by_period(:day, :recorded_at, last: DAYS_AGO)
+  end
+
+  def monthly_follow_ups
+    @monthly_follow_ups ||=
+      @current_facility
+        .patient_follow_ups(:month, last: MONTHS_AGO)
+  end
+
+  def monthly_follow_ups_by_gender
+    monthly_follow_ups
+      .group(:gender)
+  end
+
+  def controlled_visits
+    monthly_follow_ups
+      .merge(BloodPressure.under_control)
+  end
+
+  def monthly_registrations
+    @monthly_registrations ||=
+      @current_facility
+        .registered_patients
+        .group_by_period(:month, :recorded_at, last: MONTHS_AGO)
+  end
+
+  def monthly_registrations_by_gender
+    monthly_registrations
+      .group(:gender)
   end
 
-  def group_by_gender(resource_data)
-    resource_data.inject({}) do |by_gender, (gender, resource)|
-      by_gender[gender] = resource
-      by_gender
-    end
+  def all_time_follow_ups
+    @all_time_follow_ups ||=
+      @current_facility
+        .patient_follow_ups(:month)
+        .group(:gender)
+        .count
+        .each_with_object({}) do |((_, gender), count), by_gender|
+          by_gender[gender] ||= 0
+          by_gender[gender] += count
+      end
   end
 
-  def data_for_unavailable_dates(period_list)
-    period_list.map { |date| [date, 0] }.to_h
+  def all_time_registrations_by_gender
+    @current_facility
+      .registered_patients
+      .group(:gender)
   end
 
   def statistics_cache_key
-    "user_analytics/#{@current_facility.id}/#{@last_refreshed_at}"
```

Comment:
Since there's no matview business, we don't need to cache by time, we just expire every `15.minutes`